File: solutions/javascript/difference-of-squares/2/difference-of-squares.js
export class Squares {
    #n;

    constructor(/**  @type {number} n  */ n) {
        this.#n = n;
    }

    get difference() {
        return Math.abs(this.squareOfSum - this.sumOfSquares);
    }
    get sumOfSquares() {
        return (this.sum * (2 * this.n + 1)) / 3;
    }
    get squareOfSum() {
        return this.sum ** 2;
    }

    get sum() {
        return (this.n * (this.n + 1)) / 2;
    }
    get n() {
        return this.#n;
    }
}

File: solutions/julia/difference-of-squares/2/difference-of-squares.jl
"Square the sum of the first `n` positive integers"
square_of_sum(n) = sum(1:n)^2

"Sum the squares of the first `n` positive integers"
sum_of_squares(n) = n * (n + 1) * (2n + 1) ÷ 6

"Subtract the sum of squares from square of the sum of the first `n` positive ints"
difference(n) = square_of_sum(n) - sum_of_squares(n)

